System/Random.hs

```diff
@@ -1117,47 +1064,55 @@ randomIvalInteger (l,h) rng
                         (x,g') = next g
                         v' = (v * b + (fromIntegral x - fromIntegral genlo))
 
-
-bitmaskWithRejection ::
-     (RandomGen g, FiniteBits a, Num a, Ord a, Random a)
+uniformIntegerM :: (MonadRandom g m) => (Integer, Integer) -> g -> m Integer
+uniformIntegerM (l, h) gen
```

Comment:
I just converted `randomIvalIntegral` to the `MonadRandom` approach without trying to dive deeper of how the actual generation works, so if someone feels like analysing or improving the algorithm of `Integer` generation that would be great

Reply:
I (think) I understand the algorithm: it is a variant of Classic Modulo which can be badly biased. By multiplying by the arbitrary 1000 this biasing becomes a 1000 times less,

Example of biasing: in this example with 8 bits of entropy, some numbers get picked 4/3 as often, multiplying by 1000 \approxeq 10 bits, some numbers get picked 4473/4472 as often.  Of course the underlying RNG then has to do more work as well as still being (a bit) biased.
```
Data.Map.fromListWith (const (+1)) $ Prelude.map ((\x -> (x, 0)) . (`mod` 57)) [0..255]
(`mod` 57)) [0..255]
fromList [(0,4),(1,4),(2,4),(3,4),(4,4),(5,4),(6,4),(7,4),(8,4),(9,4),(10,4),(11,4),(12,4),(13,4),(14,4),(15,4),(16,4),(17,4),(18,4),(19,4),(20,4),(21,4),(22,4),(23,4),(24,4),(25,4),(26,4),(27,4),(28,3),(29,3),(30,3),(31,3),(32,3),(33,3),(34,3),(35,3),(36,3),(37,3),(38,3),(39,3),(40,3),(41,3),(42,3),(43,3),(44,3),(45,3),(46,3),(47,3),(48,3),(49,3),(50,3),(51,3),(52,3),(53,3),(54,3),(55,3),(56,3)]

fromList [(0,4473),(1,4473),(2,4473),(3,4473),(4,4473),(5,4473),(6,4473),(7,4473),(8,4473),(9,4473),(10,4473),(11,4473),(12,4473),(13,4473),(14,4473),(15,4473),(16,4473),(17,4473),(18,4473),(19,4473),(20,4473),(21,4473),(22,4473),(23,4473),(24,4473),(25,4473),(26,4473),(27,4473),(28,4473),(29,4473),(30,4473),(31,4473),(32,4473),(33,4473),(34,4473),(35,4473),(36,4473),(37,4473),(38,4473),(39,4473),(40,4472),(41,4472),(42,4472),(43,4472),(44,4472),(45,4472),(46,4472),(47,4472),(48,4472),(49,4472),(50,4472),(51,4472),(52,4472),(53,4472),(54,4472),(55,4472),(56,4472)]
```

Reply:
I should also add this function takes a generator with entropy less than 42(!) bits and makes it produce 42 bits of entropy. So if e.g. you have a generator with 1 bit of entropy then it will be invoked 42 times by this function.